### PR TITLE
Feat/order matters

### PR DESCRIPTION
**Pull request by SamuelRioTz**
https://wiki.openstreetmap.org/wiki/Relation:route
**Order matters**
we should not order it , it should be correct in osm. With this context we should only verify and detect the error for fix it on osm server

### CODE REVIEW

File: src/route_extractor.js [hunk 1]
```diff
@@ -1,105 +1,87 @@
...
+}
+const isPreviousWay = (last_way, current_way) => {
+    let res = last_way.nodes[0] == current_way.nodes[current_way.nodes.length - 1]
+    if (!res && current_way && current_way.tags && current_way.tags.oneway != "yes") {
+        revenseWay(current_way)
+        res = last_way.nodes[0] == current_way.nodes[current_way.nodes.length - 1]
+    }
+    return res
+}
 module.exports = function (route_elements, ways, assumeFirstWayIsStart) {
```
Comment (SamuelRioTz) [inline src/route_extractor.js]: i am not sure if we still need **assumeFirstWayIsStart**

File: src/extractor_error.js [hunk 2]
```diff
@@ -2,5 +2,6 @@ module.exports = {
     duplicated: "duplicated",
     not_mergeable: "not mergeable",
     undefined_street: "undefined street",
-    undefined: "undefined"
+    undefined: "undefined",
+    not_next: "streets no sequential"
```
Comment (mdoelker) [inline src/extractor_error.js]:
```suggestion
    not_next: "streets not sequential"
```

File: src/route_extractor.js (hunk 1 above)
Comment (mdoelker) [reply src/route_extractor.js]: Seems we don't

```suggestion
module.exports = function (route_elements, ways) {
```

Comment (mdoelker) [reply src/route_extractor.js]: Could you also remove the setting from the other files?

File: src/route_extractor.js [hunk 3]
```diff
@@ -1,105 +1,87 @@
 const WayPartContainer = require('./way_part_container')
 const extractor_error = require('./extractor_error')
-
+const revenseWay = (current_way) => {
```
Comment (mdoelker) [inline src/route_extractor.js]:
```suggestion
const reverseWay = (current_way) => {
```

File: src/route_extractor.js [hunk 4]
```diff
@@ -1,105 +1,87 @@
...
+    current_way.geometry = current_way.geometry.reverse()
+    current_way.nodes = current_way.nodes.reverse()
+}
+const isNextWay = (last_way, current_way) => {
+    let res = last_way.nodes[last_way.nodes.length - 1] == current_way.nodes[0]
+    if (!res && current_way && current_way.tags && current_way.tags.oneway != "yes") {
+        revenseWay(current_way)
```
Comment (mdoelker) [inline src/route_extractor.js]:
```suggestion
        reverseWay(current_way)
```

File: src/route_extractor.js [hunk 5]
```diff
@@ -1,105 +1,87 @@
...
+        res = last_way.nodes[last_way.nodes.length - 1] == current_way.nodes[0]
+    }
+    return res
+}
+const isPreviousWay = (last_way, current_way) => {
+    let res = last_way.nodes[0] == current_way.nodes[current_way.nodes.length - 1]
+    if (!res && current_way && current_way.tags && current_way.tags.oneway != "yes") {
+        revenseWay(current_way)
```
Comment (mdoelker) [inline src/route_extractor.js]:
```suggestion
        reverseWay(current_way)
```

**Pull request merged**